circular-buffer/circular_buffer_test.rb

```diff
@@ -86,6 +86,26 @@ def test_overwriting_oldest_item_in_a_full_buffer
     assert_raises(CircularBuffer::BufferEmptyException) { buffer.read }
   end
 
+  def test_forced_writes_of_nil_should_not_occupy_buffer
+    skip
+    buffer = CircularBuffer.new(2)
+    (1..2).each { |i| buffer.write String(i) }
+    buffer.write! nil
```

Comment:
This is actually an interesting question overall... it seems odd to me now that a `nil` would be treated differently than  any another value, but I see that we're already doing this for the `write` with a `nil`.

Let's do this for now, but also be open to having the discussion about what `nil` actually means in the context of a circular buffer.